model/user.go

```diff
@@ -47,6 +47,77 @@ const (
 	USER_PASSWORD_MAX_LENGTH  = 72
 )
 
+// This is a struct for a user interaction
+// A user interaction will represent one row in the results of the
+// GetUserInteractions query in store/sqlstore/user_store.go
+// **Note: Using pointers as property types allows for null values
+type UserInteraction struct {
+	// The type of interaction
+	// enum('Reaction', 'Mention', 'Reply', 'DirectMessage')
+	Interaction_Type *string `json:"Interaction_Type"`
+
+	// The username of the user that this interaction took place with
+	Username *string `json:"Username"`
+
+	// Denotes whether or not the current user was the recipient of this interaction
+	Is_Recipient *bool `json:"Is_Recipient"`
+
+	// The context in which this interaction occured (either 'Course' or the learning group type's prefix (ex. 'plg'))
+	Context *string `json:"Context"`
+
+	// The type of channel that this interaction took place in
+	// enum('P', 'O', 'D', 'G')
+	//     P = private channel
+	//     O = public channel
+	//     D = direct message (1 on 1)
+	//     G = group direct message
+	Channel_Type *string `json:"Channel_Type"`
+
+	// The name of the channel that this interaction took place in
+	Channel_Name *string `json:"Channel_Name"`
+}
+
+// This is a struct for a learning group type
+// Each learning group type is derived from personal channel names found in the config
+type LearningGroupType struct {
+	// ex. 'capstone'
+	Prefix string
+}
+
+// This is a struct for all user analytics data sets
+// This is the struct that gets returned to mattermost-redux in the users/{userId}/teams/{teamId}/useranalytics request
+// **Note: Using pointers as property types allows for null values
+type UserAnalytics struct {
+	UserInteractions   []*UserInteraction
+	UserLearningGroups []*LearningGroup
+	LearningGroupTypes []*LearningGroupType
+	// Add more datasets here
+}
+
+// This is a struct for a learning group (ex. capstone)
+// **Note: Using pointers as property types allows for null values
+type LearningGroup struct {
+	// The name of the learning group (the channel name minus the learning group type's prefix)
+	//     ex. plg-30 => 30
+	Learning_Group_Name *string
+
+	// The type of learning group (personal channel name from the config)
+	//     ex. plg
+	Learning_Group_Type *string
+
+	// The id of the private channel for this learning group
+	Channel_Id *string
+
+	// The name of the private channel for this learning group
+	Channel_Name *string
+
+	// A comma delimited list of the usernames of the members of this learning group (channel)
+	Members *string
```

Comment:
The sql query produces a comma delimited list, and this struct is for every row in the sql query. I believe I parse it into an array on the front-end, but it may be better to do that on the backend, although it would require some work.